rest-api-spec/src/main/resources/rest-api-spec/test/search.aggregation/360_date_histogram.yml

```diff
@@ -109,6 +109,19 @@ setup:
       version: " - 2.16.99"
       reason: "Bugfix not backported to 2.16"
 
+  - do:
+      indices.create:
+        index: dhisto-agg-w-query
+        body:
+          settings:
+            number_of_shards: 1
+            number_of_replicas: 0
+            refresh_interval: -1
```

Comment:
Why do we need to explicitly create the index with custom settings? Can't we just `forcemerge` regardless and it should work?

Reply:
You are right in that, but i'd rather we prevent the creation of multiple segments with the explicit refresh and not rely on the forcemerge.

Reply:
I think the [PR](https://github.com/opensearch-project/OpenSearch/pull/16067) on `main` is already merged. Tagging @finnegancarroll to share his thoughts here.

Reply:
> You are right in that, but i'd rather we prevent the creation of multiple segments with the explicit refresh and not rely on the forcemerge.

While I don't mind that as such, but we should be good with either. Also, why limit the `number_of_shards` to 1, we can leave that to be the default. Just to ensure we don't have any regression in future pertaining to multiple shards.

Reply:
Setting shards to 1 made sense to me since we are asserting on a shard specific stat.
`  - match: { profile.shards.0.aggregations.0.debug.optimized_segments: 1 }`

The force merge is redundant with `refresh_interval: -1`. There are several places with this pattern, can I fix them all with a follow up pr keeping the index settings and removing the force merge operations?

Reply:
Follow up pr to remove these redundant force merge steps - https://github.com/opensearch-project/OpenSearch/pull/16144